mod/ArchConsoleManager.cs

```diff
@@ -0,0 +1,189 @@
+﻿using OWML.Common;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace ArchipelagoRandomizer
+{
+    /// <summary>
+    /// Class that handles adding text to the console
+    /// </summary>
+    public class ArchConsoleManager : MonoBehaviour
+    {
+        private GameObject pauseConsole;
+        private GameObject pauseConsoleContent;
+        private GameObject pauseConsoleViewer;
+        private GameObject gameplayConsole;
+        private GameObject consoleEntry => Randomizer.Assets.LoadAsset<GameObject>("ConsoleText");
+        private List<string> consoleMasterList;
+        private List<GameObject> pauseConsoleEntries;
+        private List<GameObject> gameplayConsoleEntries;
```

Comment:
...
Copying old code from another project, where I was using a ton of different types of layout elements, and some of these hacky workarounds were required, lol.
I think I did initially try this, but I was getting an issue where the text was saying it had 0 height unless there was another object in the layout. However, I think I've since fixed this.

Reply:
Oh I think a big reason for it was to make it easier to limit the gameplay console to six entries.

Reply:
I think I'm missing something important then. Why would it be easier to limit the size of a `List<GameObject>` than to limit the size of a `List<string>`?

Reply:
Because I was thinking of it as a single large string rather than a list of strings. But if we're still using a List, then yeah it wouldn't be any harder.

Reply:
Oh, yes, strong agree the source of truth should be a real `List<>` instead of just one big `string`.

I thought my InGameConsole code was already using a `List<string>` for storage and joining them with newlines in Update.